### ISSUE TITLE
`net` module: no response or error when 'content-encoding' is incorrect and session is not persistent

### ISSUE THREAD

**Issue by arantes555** [electron/electron#13587]
* Electron Version: all versions >= 1.7 (tested on 1.7.0, 1.7.15, 1.8.7, 2.0.4)
* Operating System (Platform and Version): macOS 10.13.5
* Last known working Electron version: 1.6.18

**Expected Behavior**
When an endpoint responds with an incorrect `'content-encoding'`, the `response` object from the `net` request should emit a `net::ERR_CONTENT_DECODING_FAILED` error.

**Actual behavior**
When making a request to an endpoint that responds with an incorrect `'content-encoding'`:
- With a persistent (or the default) session, it behaves normally (emitting a `net::ERR_CONTENT_DECODING_FAILED` error)
- With a non-persistent session, nothing happens. We never get the error.

**To Reproduce**

```sh
git clone https://github.com/arantes555/electron-bug-net-content-encoding.git -b master
cd electron-bug-net-content-encoding

# electron@2.0
npm install
npm start # see that the last request gives a timeout


# electron@1.6
git checkout electron-1.6
npm install
npm start # see that all requests behave as intended
```

 **Additional information**

It seems somewhat related to https://github.com/electron/electron/issues/8867 , but it is definitely not the same issue.

**Comment by arantes555** [comment]
The issue still exists with electron@3

The exemple repo has been updated to show it.

```bash
# electron@3.0
git checkout electron-3
npm install
npm start # see that the last request gives a timeout
```

**Comment by sofianguy** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron v4.2.x or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons. This issue will be closed 7 days from now if there is no response.

Thanks in advance! Your help is appreciated.

**Comment by arantes555** [comment]
Hello @sofianguy.

I just checked electron 4 and 5. Here are the results:

- For electron@4.2.9, the behaviour is even worse. There always is a timeout when the content-encoding is invalid, even if the session is persistent. `[Error: net::ERR_CONTENT_DECODING_FAILED]` is never emitted.
- For electron@5.0.9, it's the same as electron@4. The behaviour is worse than with electron 2 & 3.

**Comment by arantes555** [comment]
I can confirm the behaviour is still the same with electron6.

With electron-7, the problem seems fixed, but a lot of other `net` problems make it basically unsuable

**Comment by arantes555** [comment]
(this issue needs the `component/net` tag btw)

**Comment by nornagon** [comment]
Electron 6.x is no longer supported. @arantes555, a lot of work has been done to fix issues in the net module in Electron 7+. If you're still seeing other problems in the net module in recent releases of Electron, please open new issues for them :)

**Comment by arantes555** [comment]
@nornagon I can confirm the bug is fixed in versions >= 7.0.0

Thanks
